app/controllers/global_activity_analytics_controller.rb

```diff
@@ -0,0 +1,10 @@
+class GlobalActivityAnalyticsController < ApplicationController
+  def show
+    begin
+      UpdateGlobalActivityAnalytics.new.perform
+    rescue => e
+      Rollbar.error(e)
+    end
```

Comment:
I'm pretty sure this `rescue` block is unnecessary since Rollbar will capture all exceptions automatically.

Reply:
Answering to the inlined question and this one:
> Is this supposed to automatically refresh every 24 hours? If so, I don't see how that happens.

With current implementation update cache is doing *on demand* but not more than once in 24h. The cache update is triggered right from this controller action. How does it to work:
- on the very first request the cache is outdated (it's `GlobalActivityAnalytics.instance.updated_at` field is later than 24h ago, see `GlobalActivityAnalytics.instance`  implemetation)
- when a mobile client performs the request the two cases are possible:
-- cache is up to date: `GlobalActivityAnalytics.instance.updated_at` <= 24h.
   In this case `UpdateGlobalActivityAnalytics.new.perform` will do nothing (see guard clause in `UpdateGlobalActivityAnalytics#perform` - `return if @analytics.actual?`), i.e. `rescue` won't be evaluated
-- cache is outdated: `GlobalActivityAnalytics.instance.updated_at` > 24h.
   In this case `UpdateGlobalActivityAnalytics.new.perform` will try to update the counters by requesting analytics API. Since it's a network call various exceptions are possible: network connection error, timeouts, TLS, etc.
   I don't want to expose these exceptions to the end client and want to return some data always. But these exceptions is an important information for developers, and most probably they would need to react on them. Therefore, it catches these exceptions and returns the previously cached results to the end-user. To implement the monitoring for developers it logs the exception into Rollbar. It's done with `rescue`

Reply:
Got it.  Thanks.